### PR TITLE
improvement(executor): subflows, hitl handling cleanup 

### PR DESCRIPTION
**Pull request by icecrasher321**
## Summary

Refactors parallel batching onto explicit DAG control flow while hardening nested subflow execution, pause/resume, run-from-block, and output resolution edge cases.

## Type of Change
- [x] Other: Code quality

## Testing
Tested manually

## Checklist
- [x] Code follows project style guidelines
- [x] Self-reviewed my changes
- [x] Tests added/updated and passing
- [x] No new warnings introduced
- [x] I confirm that I have read and agree to the terms outlined in the [Contributor License Agreement (CLA)](./CONTRIBUTING.md#contributor-license-agreement-cla)

### CODE REVIEW

**Comment by icecrasher321** [comment]
bugbot run

**Comment by icecrasher321** [comment]
@greptile

**Comment by icecrasher321** [comment]
@greptile

**Comment by icecrasher321** [comment]
bugbot run

File: apps/sim/executor/execution/state.ts [hunk 1]
```diff
@@ -51,23 +63,33 @@ export class ExecutionState implements BlockStateController {
...
-      if (suffixedOutput !== undefined) {
-        return suffixedOutput
+      const scopedOutput = this.getScopedBlockOutput(blockId, currentNodeId)
+      if (scopedOutput !== undefined) {
+        return scopedOutput
+      }
+
+      if (extractOuterBranchIndex(currentNodeId) !== undefined) {
+        return undefined
+      }
```
[the remark this answers is not in the document]
Comment (icecrasher321) [reply apps/sim/executor/execution/state.ts:84]: the fix is at the resolver level -- already done

**Comment by icecrasher321** [comment]
@greptile

**Comment by icecrasher321** [comment]
@greptile

**Comment by icecrasher321** [comment]
bugbot run

**Comment by icecrasher321** [comment]
bugbot run

**Comment by icecrasher321** [comment]
@greptile

File: apps/sim/executor/execution/edge-manager.ts [hunk 2]
```diff
@@ -170,45 +191,39 @@ export class EdgeManager {
...
+    if (
+      sourceSubflowType &&
+      sentinelSubflowType &&
+      sourceSubflowType === sentinelSubflowType &&
+      sourceSubflowId &&
+      sentinelSubflowId &&
+      sourceSubflowId === sentinelSubflowId
+    ) {
       return true
+    }
```
[the remark this answers is not in the document]
Comment (icecrasher321) [reply apps/sim/executor/execution/edge-manager.ts:208]: isEnclosingSentinel should only wake the sentinel for the same immediate subflow. For a condition inside a loop nested in a parallel:

The condition dead-end should wake the inner loop’s sentinel-end.
The loop sentinel-end then evaluates continuation/exit.
If it exits, its LOOP_EXIT edge activates the parent parallel sentinel-end.
Letting the condition directly wake the outer parallel sentinel would bypass loop semantics and could aggregate/finish the parent branch too early. So the subflowType equality check is intentional here, not a regression.

**Comment by icecrasher321** [comment]
bugbot run

File: apps/sim/executor/execution/engine.ts [hunk 3]
```diff
@@ -412,6 +415,8 @@ export class ExecutionEngine {
     }
 
     if (output._pauseMetadata) {
+      await this.nodeOrchestrator.handleNodeCompletion(this.context, nodeId, output)
+
```
[the remark this answers is not in the document]
Comment (icecrasher321) [reply apps/sim/executor/execution/engine.ts:406]: it should not survive to final aggregation after resume. human-in-the-loop-manager.ts explicitly scans parallelExecutions.branchOutputs for the matching _pauseMetadata.contextId and replaces that entry with the merged human response before resuming.

**Comment by icecrasher321** [comment]
bugbot run

**Comment by icecrasher321** [comment]
@greptile

**Comment by icecrasher321** [comment]
bugbot run

**Comment by icecrasher321** [comment]
bugbot run

**Comment by icecrasher321** [comment]
@greptile
